quit/basicfunction.py

```diff
@@ -0,0 +1,138 @@
+import typing
+
+import numpy as np
+from scipy import linalg
+
+Array1D = typing.Union[np.ndarray, typing.Sequence[complex]]
+
+
+def ket(psi: Array1D) -> np.ndarray:
+    """Create a ket |psi> as a vertical vector.
+
+    :param psi: Sequence of complex numbers.
+    :return: A vertical vector as numpy array.
+    """
+    psi = np.asarray(psi)
+
+    if len(psi.squeeze().shape) != 1:
+        raise ValueError("Wrong parameter psi.")
+    return psi.reshape((-1, 1))
+
+
+def basis(dim: int, index: int) -> np.ndarray:
+    """Create computational basis of dimension dim.
+
+    :param dim: Dimension of vector space.
+    :param index:
+    :return: A basis vector |d, i>.
+    """
+    if index >= dim:
+        raise Exception("Parameter index out of range.")
```

Comment:
Using the base `Exception` class is too broad and may make it hard to handle errors by client code. I suggest raising `ValueError `instead.